src/Microsoft.DotNet.Configurer/DotnetFirstRunConfiguration.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class DotnetFirstRunConfiguration
+    {
+        public bool GenerateAspNetCertificate { get; set; }
+
+        public bool PrintTelemetryMessage { get; set; }
+
+        public bool SkipFirstRunExperience { get; set; }
```

Comment:
I think it is time to make "FirstRunExperience" more specific in code. Since print note could also be firstrunexperience. Just call it, expendCacheAndSetEnvAnd...

it's ugly, but a firstrunexperience field in a whole logic about firstrunexperience almost has no meaning.

Reply:
Turning first run off turns everything off, not just the messages.